backend/code/m-booking/src/test/java/com/mbooking/service/UserServiceIntegrationTest.java

```diff
@@ -0,0 +1,245 @@
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.mbooking.dto.EditProfileDTO;
+import com.mbooking.dto.UserDTO;
+import com.mbooking.exception.ApiAuthException;
+import com.mbooking.exception.ApiException;
+import com.mbooking.model.Admin;
+import com.mbooking.model.Authority;
+import com.mbooking.model.Customer;
+import com.mbooking.model.User;
+import com.mbooking.repository.AuthorityRepository;
+import com.mbooking.repository.CustomerRepository;
+import com.mbooking.repository.UserRepository;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("test_h2")
+public class UserServiceIntegrationTest {
+
+	@Autowired
+	private UserService userService;
+
+	@Autowired
+	private UserRepository userRepo;
+
+	@Autowired
+	private CustomerRepository customerRepo;
+
+	@Autowired
+	private PasswordEncoder passwordEncoder;
+
+	@Autowired
+	private AuthorityRepository authorityRepo;
+
+	@Autowired
+	private EmailSenderService emailSenderService;    
+
+	
+	
+	/*@Before
+	public void setUp() {
+		Customer cus = new Customer();
+		cus.setEmail("email");
+		cus.setEmailConfirmationId("eid");
+		cus.setFirstname("name");
+		cus.setLastname("lastname");
+		cus.setEmailConfirmed(false);
+		cus.isEmailConfirmed();
+
+
+	       Authority roleRegistered = new Authority((long) -2,"ROLE_ADMIN");
+	        authorityRepo.save(roleRegistered);
+
+	        userRepo.save(cus);
+	      
+
+	    }*/
+		
+	
+	
+	
+	
+	@Test(expected = DataIntegrityViolationException.class)
+	public void testRegister() {
+		UserDTO userDTO = new UserDTO();
+		userDTO.setId(1L);
+		userDTO.setFirstname("user5");
+		userDTO.setLastname("user5");
+		userDTO.setEmail("email2@gmail.com");
+		userDTO.setPassword("user");
+		
+		Customer customer = new Customer();
+		customer.setEmail(userDTO.getEmail());
+		customer.setFirstname(userDTO.getFirstname());
+		customer.setLastname(userDTO.getLastname());
+		customer.setPassword(passwordEncoder.encode(userDTO.getPassword()));
+		
+		UserDTO rez = userService.register(userDTO);
+		
+		
+		emailSenderService.sendSimpleMessage(customer.getEmail(),"sub", "tex");
+		
+		
+		assertEquals(customer.getEmail(), rez.getEmail());
+		assertEquals(customer.getFirstname(), rez.getFirstname());
+		assertEquals(customer.getLastname(), rez.getLastname());
+		
+		customerRepo.save(customer);
+		authorityRepo.findByName("ROLE_CUSTOMER");
+	}
+
+	@Test(expected = ApiException.class)
+	public void testRegister1() {
```

Comment:
Please don't name test like this. We were told to follow the given.._when..._then convention
givenExistingCredentials_whenRegistering_throwException or something like that

Reply:
okej